File: C#/242_ValidAnagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        
        if (s.Length != t.Length)
            return false;
        
        Dictionary<char,int> m = new Dictionary<char,int>();
        
        for (int i = 0; i < s.Length; i++){
            
            if (m.ContainsKey(s[i])){
                m[s[i]]++;
            }else 
                m.Add(s[i], 1);
            
            if (m.ContainsKey(t[i])){
                m[t[i]]--;
            }else 
                m.Add(t[i], -1);
        }
        
        foreach(KeyValuePair<char,int> kvp in m){
            if (kvp.Value != 0)
                return false;
        }
        
        return true;
    }
}

File: Python/242_ValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        letters = dict()
        for letter in s:
            if letter in letters:
                letters[letter] += 1
            else:
                letters[letter] = 1
        
        for letter in t:
            if letter not in letters:
                return False
            letters[letter] -= 1
            if letters[letter] == 0:
                letters.pop(letter)
        
        if len(letters) > 0:
            return False
        
        return True
